components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/DashboardMetadataProvider.java

```diff
@@ -108,4 +109,7 @@ void updateDashboardRoles(String user, String dashboardUrl, Map<String, List<Str
      * @throws DashboardException If an error occurred while reading or processing dashboards
      */
     DashboardArtifact exportDashboard(String dashboardUrl) throws DashboardException;
+
+    ReportConfigs getReportGenerationConfigurations() throws DashboardException;
```

Comment:
Instead, lets return the full `DashboardConfigurations` object. No need to `throws DashboardException`

Reply:
Done